### PR TITLE
Plot data

### PR DESCRIPTION
**Pull request by Jeremie-Ch**
* **Please check if the PR fulfills these requirements**
- [ ] The commit messages follows our guidelines (explicit, in english)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bug fix:
- [x] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:
I modified several classes' __init__ (Circle2D, LineSegment2D, Line2D) such that they can take a list [x,y] as input, though a Point2D is still accepted. I found it annoying to instanciate a Point2D every time I had to create a primitive so now whenever a list [x,y] is passed as input, it is converted into a Point2D by the constructor.

### CODE REVIEW

File: volmdlr/edges.py [hunk 1]
```diff
@@ -247,6 +245,10 @@ class Line2D(Line):
     """
 
     def __init__(self, point1, point2, *, name=''):
+        if isinstance(point1, list):
```
Comment (GhislainJ) [inline volmdlr/edges.py:248]: isinstance(obj, list) isn't robust enough to check for an object to be a sequence. If we are to be less strict on the way to provide data to init, tuples should be ok as well. dessia_common defines a way to check it. Consider using is_sequence instead

```suggestion
        if dc.is_sequence(point1):
```

File: volmdlr/edges.py [hunk 2]
```diff
@@ -247,6 +245,10 @@ class Line2D(Line):
...
+            point1 = volmdlr.Point2D(point1[0], point1[1])
+        if isinstance(point2, list):
```
Comment (GhislainJ) [inline volmdlr/edges.py:250]: Same as previous comment : 

```suggestion
        if dc.is_sequence(point2):
```

File: volmdlr/edges.py [hunk 3]
```diff
@@ -642,29 +644,33 @@ def __init__(self, degree: int, control_points: List[volmdlr.Point2D],
...
 
 
-
 class LineSegment2D(LineSegment):
     """
     Define a line segment limited by two points
     """
 
     def __init__(self, start, end, *, name=''):
+        if isinstance(start, list):
```
Comment (GhislainJ) [inline volmdlr/edges.py:653]: Same as previous comment on sequence check : 

```suggestion
        if dc.is_sequence(start):
```

File: volmdlr/edges.py [hunk 4]
```diff
@@ -642,29 +644,33 @@ def __init__(self, degree: int, control_points: List[volmdlr.Point2D],
...
+            start = volmdlr.Point2D(start[0], start[1])
+        if isinstance(end, list):
```
Comment (GhislainJ) [inline volmdlr/edges.py:655]: Same as previous comment on sequence check : 

```suggestion
        if dc.is_sequence(end):
```

File: volmdlr/wires.py [hunk 5]
```diff
@@ -1373,15 +1373,16 @@ def area(self):
...
-
 class Circle2D(Contour2D):
     _non_serializable_attributes = ['internal_arcs', 'external_arcs',
                                     'polygon', 'straight_line_contour_polygon',
                                     'primitives', 'basis_primitives']
 
-    def __init__(self, center: volmdlr.Point2D, radius: float, name: str = ''):
-        self.center = center
+    def __init__(self, center, radius: float, name: str = ''):
+        if isinstance(center, list):
```
Comment (GhislainJ) [inline volmdlr/wires.py:1382]: Same as previous comment on sequence check : 

```suggestion
        if dc.is_sequence(center):
```

**Comment by GhislainJ** [comment]
I am not a huge fan of being less strict on provided arguments types as it is more prone to usage errors and is less easy to maintain. But for volmdlr purposes, I prefer letting @masfaraud judge if these changes are ok.

In any case, I let some comments and suggestions for sequence checking. Only looking for an object to be an instance of list might lead to unefined case as it could behave like a sequence without being a list (tuples,...)

**Comment by masfaraud** [comment]
Good idea for usability but bad idea for performance: checking data each time a point is instantiated is a not efficient.

**Pull request closed without merging**